python/ray/tests/test_failure_4.py

```diff
@@ -752,6 +752,56 @@ def get_result(self, a):
         assert isinstance(e, ray.exceptions.RayActorError)
 
 
+@pytest.mark.parametrize(
+    "ray_start_cluster",
+    [
+        {
+            "num_cpus": 1,
+            "num_nodes": 1,
+        }
+    ],
+    indirect=True,
+)
+def test_shows_both_user_exception_system_error_same_time(ray_start_cluster):
+    @ray.remote(max_calls=1)
+    def f():
+        raise Exception("this is a exception")
+
+    with pytest.raises(Exception):
+        ray.get(f.remote())
+
+    # Wait for the task info to be propagated.
+    import time
+
+    time.sleep(1)
+
+    tasks = list_tasks(filters=[("name", "=", "f")], detail=True)
+    assert len(tasks) == 1, tasks
+    task = tasks[0]
+    assert task["state"] == "FAILED"
+    assert task["error_type"] == "TASK_EXECUTION_EXCEPTION"
+    # The error message should look like below (modulo line breaks), and we compare
+    # without the stacktrace:
+    #
+    # User exception:
+    # ray::f() (pid=70293, ip=127.0.0.1)
+    #   File "<YOUR_RAY_DIR>/python/ray/tests/test_exit_observability.py", line 465,
+    # in f
+    #     raise Exception("this is a exception")
+    # Exception: this is a exception
+    #
+    # System error:
+    # IntentionalSystemExit: Worker exits with an exit code 0. max_call has reached,
+    # max_calls: 1
+    error_message = task["error_message"]
+    assert error_message.startswith("User exception:\nray::f()"), error_message
+    assert error_message.endswith(
+        "Exception: this is a exception\n\nSystem error:\n"
+        "IntentionalSystemExit: Worker exits with an exit "
+        "code 0. max_call has reached, max_calls: 1"
```

Comment:
Not related to this PR, but could you update the message for this error? I think it could be clearer, like:

"Exited because worker reached max_calls={n} for this method."

Reply:
done.